// Copyright (c) 2017-2019, NVIDIA CORPORATION.  All rights reserved.

/**
* Protocol buffer definition for specifying optimizer parameters.
*/

syntax = "proto3";

import "nvidia_tao_tf1/cv/common/proto/sgd_optimizer_config.proto";
import "nvidia_tao_tf1/cv/common/proto/adam_optimizer_config.proto";
import "nvidia_tao_tf1/cv/common/proto/rmsprop_optimizer_config.proto";

message OptimizerConfig {
	oneof optimizer {
		AdamOptimizerConfig adam = 1;
        SGDOptimizerConfig sgd = 2;
        RMSpropOptimizerConfig rmsprop = 3;
	}
}
